<?xml version="1.0" ?>
<launch>

  <arg name="map_file" default="$(find pyCAS)/domains/CDB_robot/maps/campus_1_test.yaml" />
  <arg name="model" default="burger"/>
  <arg name="move_forward_only" default="false"/>
  <arg name="x_pos" default="-1.0"/>
  <arg name="y_pos" default="1.0"/>
  <arg name="z_pos" default="0.0"/>
  <arg name="world" default="$(find pyCAS)/gazebo/worlds/campus_4.world"/>

  <!-- playground to get the planner to work-->
  <node pkg="pyCAS" type="test_gazebo.py" name="test_gazebo" output="screen"/>

  <!-- include gazebo_ros launcher -->

  <include file="$(find gazebo_ros)/launch/empty_world.launch">
          <arg name="world_name" value= "$world" />
          <arg name="debug" value="false" />
          <arg name="gui" value="true" />
          <arg name="paused" value="false" />
          <arg name="use_sim_time" value="true" />
          <arg name="headless" value="false"/>
  </include>

  <!-- turtlebot 3-->

  <param name="robot_description" command="$(find xacro)/xacro --inorder $(find turtlebot3_description)/urdf/turtlebot3_$(arg model).urdf.xacro" />
  
  <node pkg="gazebo_ros" type="spawn_model" name="spawn_urdf" args="-urdf -model turtlebot3_$(arg model) -x $(arg x_pos) -y $(arg y_pos) -z $(arg z_pos) -param robot_description" />

  <include file="$(find turtlebot3_bringup)/launch/turtlebot3_remote.launch">
    <arg name="model" value="$(arg model)" />
  </include>

  <param name="tb3_model" value="burger"/>


  <!-- localization-->
  <include file="$(find turtlebot3_navigation)/launch/amcl.launch"/> 

  <!-- visualization and mapping -->
  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find pyCAS)/rviz/default.rviz" />

  <node pkg="map_server" name="map_server" type="map_server" args="$(arg map_file)"/>

  <!-- Move_base node is the ros planner  -->
  <include file="$(find turtlebot3_navigation)/launch/move_base.launch">
    <arg name="model" value="$(arg model)" />
    <arg name="move_forward_only" value="$(arg move_forward_only)"/>
    <rosparam file="$(find turtlebot3_navigation)/param/costmap_common_params.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find turtlebot3_navigation)/param/costmap_common_params.yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find turtlebot3_navigation)/param/local_costmap_params.yaml" command="load" />
    <rosparam file="$(find turtlebot3_navigation)/param/global_costmap_params.yaml" command="load" />
    <rosparam file="$(find turtlebot3_navigation)/param/dwa_local_planner_params.yaml" command="load" />
    <remap from="cmd_vel" to="mobile_base/commands/velocity"/>
  </include>
</launch>
